cmd/influxd/launcher/storage_test.go

```diff
@@ -255,6 +256,226 @@ func TestLauncher_DeleteWithPredicate(t *testing.T) {
 	}
 }
 
+func TestLauncher_FluxCardinality(t *testing.T) {
+	l := launcher.RunAndSetupNewLauncherOrFail(ctx, t)
+	defer l.ShutdownOrFail(t, ctx)
+
+	// Run a query without any data on the server - should return 0 and not crash.
+	query := `import "influxdata/influxdb"
+	influxdb.cardinality(
+		bucket: "BUCKET",
+		start: 2000-01-01T00:00:00Z,
+		stop: 2000-01-02T00:00:00Z,
+		predicate: (r) => true
+	)`
+
+	exp := `,result,table,_value` + "\r\n" +
+		`,_result,0,0` + "\r\n\r\n"
+
+	buf, err := http.SimpleQuery(l.URL(), query, l.Org.Name, l.Auth.Token)
+	if err != nil {
+		t.Fatalf("unexpected error querying server: %v", err)
+	} else if diff := cmp.Diff(string(buf), exp); diff != "" {
```

Comment:
Preference would be for `require.Equal` instead of `cmp.Diff` + `t.Fatal`

Reply:
Updated; see above.